### ISSUE TITLE
Validation error in Workbench

### ISSUE THREAD

**Issue by acbentley** [specify/specify7#1533]
**Describe the bug**
I am getting a validation error when validating the Tulsa Series 1 dataset in the workbench

**To Reproduce**
1. Go to Workbench
2. Click on Tulsa Series 1 dataset
3. Click validate

**Expected behavior**

**Screenshots**
![Capture](https://user-images.githubusercontent.com/7501989/167899091-0ae2ff2b-115c-4875-a351-6b2e1f5ee331.JPG)

**Desktop (please complete the following information):**
 - OS: [e.g. Windows 10]
 - Browser [e.g. Chrome, Safari]
 - Specify 7 Version [e.g. 7.6.1]

**Reported By**
Name of your institution

**Additional context**

**Comment by benanhalt** [comment]
The data mapper has a field which is not fully mapped:

![image](https://user-images.githubusercontent.com/287254/167926602-d1bbbf5b-6444-4a1f-9f1e-fcd249ddeb52.png)

but it is including the partially mapped field in the upload plan in a way that confuses the uploader:
```
{
  "baseTableName": "collectionobject",
  "uploadable": {
    "uploadTable": {
      "wbcols": {
        "catalognumber": "Catalog #",
        "catalogeddate": "Catalog date",
        "fieldnumber": "Field #"
      },
<SNIP>
        "preparations": [
          {
            "wbcols": {
              "countamt": "Count",
              "storage": "Storage"
            },
            "static": {},
            "toOne": {
              "preptype": {
                "mustMatchTable": {
                  "wbcols": {
                    "name": "Prep type"
                  },
                  "static": {},
                  "toOne": {},
                  "toMany": {}
                }
<SNIP>
```

**Comment by benanhalt** [comment]
Since preparation -> storage is a relationship but is being included in the regular `wbcols` part of the plan, it is expecting storageid values, which is why it is upset when it gets strings instead of integers.

**Comment by benanhalt** [comment]
Okay. The uploader should no longer be confused by such mappings, but the mapper UI should probably not allow them.

**Comment by maxxxxxdlp** [comment]
That data set was probably created in the current release.
New release throws out unmapped fields.

![Screen Shot 2022-05-11 at 16 53 49](https://user-images.githubusercontent.com/40512816/167954044-647bb91e-5c12-4934-a9b2-e9d00c9269f9.png)

**Comment by maxxxxxdlp** [comment]
> Okay. The uploader should no longer be confused by such mappings, but the mapper UI should probably not allow them.

Does it mean that WB now supports mapping directly to the ID column?

**Comment by benanhalt** [comment]
Sort of. But I wouldn't rely on it.

**Comment by benanhalt** [comment]
The problem is the `treeNodeCols` are using the mixed case versions of the column names. See the "commonName" and "yesNo*" fields below. I just pass them straight to Django which is expecting all lowercase. I can coerce them to lowercase on the server, but I think the corresponding `wbCols` mappings are using the lowercase versions. Maybe it is better to just use the Django convention everywhere.

```json
{
  "baseTableName": "taxon",
  "uploadable": {
    "treeRecord": {
      "ranks": {
        "Division": {
          "treeNodeCols": {
            "name": "division"
          }
        },
        "Class": {
          "treeNodeCols": {
            "name": "class"
          }
        },
        "Subclass": {
          "treeNodeCols": {
            "name": "subclass"
          }
        },
        "Order": {
          "treeNodeCols": {
            "name": "order"
          }
        },
        "Family": {
          "treeNodeCols": {
            "name": "family",
            "text20": "scfamily"
          }
        },
        "Subfamily": {
          "treeNodeCols": {
            "name": "subfamily"
          }
        },
        "Tribe": {
          "treeNodeCols": {
            "name": "tribe"
          }
        },
        "Genus": {
          "treeNodeCols": {
            "name": "genus",
            "text20": "scgenus"
          }
        },
        "Species": {
          "treeNodeCols": {
            "name": "species",
            "author": "author",
            "commonName": "Species common name",
            "text9": "Species name",
            "text1": "origin",
            "yesNo1": "reported",
            "yesNo2": "accepted",
            "yesNo3": "confirmed",
            "yesNo4": "excluded",
            "yesNo5": "binomonly",
            "text20": "scspecies",
            "text19": "scauthor",
            "text18": "scpage",
            "yesNo6": "needsystem",
            "yesNo7": "needfield",
            "yesNo8": "problem",
            "yesNo9": "wetland",
            "yesNo10": "dermatitis",
            "yesNo11": "intpoison",
            "text2": "cult",
            "text17": "indicator1",
            "text5": "schabitat",
            "text14": "source",
            "text15": "srcedate",
            "timestampModified": "editdate",
            "text3": "scname",
            "integer1": "numbercos"
          }
        },
        "Subspecies": {
          "treeNodeCols": {
            "name": "subtaxon1",
            "author": "subauthor1",
            "commonName": "subsp. common name",
            "text9": "subsp.name"
          }
        },
        "Variety": {
          "treeNodeCols": {
            "name": "subtaxon2",
            "author": "subauthor2",
            "commonName": "var. common name",
            "text9": "var. name"
          }
        }
      }
    }
  }
}
```

**Comment by maxxxxxdlp** [comment]
Ah, good catch!
Fixed in e550faae
